conan/internal/runner/podman.py

```diff
@@ -0,0 +1,276 @@
+        finally:
+            if self.container:
+                error_prefix = 'ERROR: ' if error else ''
+                _podman_info(f'{error_prefix}Stopping container', error)
+                self.container.stop()
+                if self.remove:
+                    _podman_info(f'{error_prefix}Removing container', error)
+                    self.container.remove()
+
+    def build_image(self):
+        dockerfile_file_path = self.dockerfile
+        if os.path.isdir(self.dockerfile):
+            for df in ['Containerfile', 'Dockerfile']:
+                dockerfile_file_path = os.path.join(self.dockerfile, df)
+                if os.path.exists(dockerfile_file_path): break
+        build_path = self.podman_build_context or os.path.dirname(dockerfile_file_path)
+        ConanOutput().highlight(f"Container recipe path: '{dockerfile_file_path}'")
+        ConanOutput().highlight(f"Container build context: '{build_path}'\n")
+        _, podman_build_logs = self.podman_client.images.build(
+            dockerfile=dockerfile_file_path,
+            path=build_path,
+            tag=self.image,
+            buildargs=self.configfile.build.build_args,
+            cache_from=self.configfile.build.cache_from,
+        )
+        for chunk in podman_build_logs:
+            for line in chunk.decode("utf-8").split('\r\n'):
+                if line:
+                    stream = json.loads(line).get('stream')
+                    if stream:
+                        ConanOutput().status(stream.strip())
+
+    def run_command(self, command, workdir=None, log=True):
+        workdir = workdir or self.abs_podman_path
+        if log:
+            _podman_info(f'Running in container: "{command}"')
+        _, exec_output = self.container.exec_run(f"/bin/bash -c '{command}'", stream=True, workdir=workdir, demux=True)
+        stderr_log, stdout_log = '', ''
+        try:
+            for (stdout_out, stderr_out) in exec_output:
+                if stdout_out is not None:
+                    stdout_log += stdout_out.decode('utf-8', errors='ignore').strip()
+                    if log:
+                        ConanOutput().status(stdout_out.decode('utf-8', errors='ignore').strip())
+                if stderr_out is not None:
+                    stderr_log += stderr_out.decode('utf-8', errors='ignore').strip()
+                    if log:
+                        ConanOutput().status(stderr_out.decode('utf-8', errors='ignore').strip())
+        except Exception as e:
+            if platform.system() == 'Windows':
+                import pywintypes
+                if isinstance(e, pywintypes.error):
+                    pass
+            else:
+                raise e
+        #exit_metadata = self.docker_api.exec_inspect(exec_instance['Id'])
+        #if exit_metadata['Running'] or exit_metadata['ExitCode'] > 0:
+        #    raise RunnerException(command=command, stdout_log=stdout_log, stderr_log=stderr_log)
```

Comment:
There seems to be no way to check for the command exit code while using `stream=True`.